src/app/core/services/auth.service.ts

```diff
@@ -33,6 +33,9 @@ export enum AuthState {
  * updating the application state with regards to authentication.
  */
 export class AuthService {
+  private static readonly DEFAULT_HAS_PRIVATE_PERMISSION = false;
```

Comment:
Small nit - might want to change the constant name here

Reply:
I changed the constant name to `DEFAULT_HAS_PERMISSION_TO_PRIVATE_REPOS`

Reply:
hmm what i meant is that the constant name sort of implies that the constant value is `true`, but it's actually `false`. maybe something like `DEFAULT_DOES_NOT_HAVE_PRIVATE_REPO_PERMISSION` might work

Reply:
To avoid the name being interpreted as a boolean question, I have renamed it to `NO_PERMISSION_TO_PRIVATE_REPOS`

Reply:
In my opinion I get where @luminousleek is coming from. Default should indicate its default state in the name following the word DEFAULT.
However, this is confusing since developers might mistake it as representing a boolean variable which means "No private repo access", when really it is used as "has private repo access".
I think a better name would be just HAS_PERMISSION_TO_PRIVATE_REPOS. why the DEFAULT at all when it is implied already by static readonly.

Reply:
Without the word `DEFAULT`, it might be mistaken that the auth service will never have access to private repos, while in fact, this is only the default settings when the app skips manual login. The app would still have access to private repos if the user chooses to (with manual login).

Phrasing the name as a boolean question (i.e. `HAS_PERMISSION_TO_PRIVATE_REPOS`) is more applicable when it is a boolean variable that can take up both values (i.e. `true` and `false`) in different scenarios, but in this case, it can only be `false`. Some might infer the variable name as the auth service always have access to private repos.

I would still be more inclined to use `NO_PERMISSION_TO_PRIVATE_REPOS`, because it really means "no private repo access", and is meant to be used only when the app has no private repo access. Unless there is a good way to incorporate the word `DEFAULT` into the constant name when phrasing it as a boolean question.

Reply:
Late to the discussion, I would prefer `DEFAULT_NO_PERMISSION_TO_PRIVATE_REPOS` or similar. Default keyword is important to show that it is by default

Reply:
Changed to `DEFAULT_NO_PERMISSION_TO_PRIVATE_REPOS`!